.github/workflows/main.yaml

```diff
@@ -297,15 +325,10 @@ jobs:
         include: ${{ fromJson(needs.initialize.outputs.includes) }}
         config: [devel]
     steps:
-      - name: Check Status
-        if: ${{ ! (needs.on-device-test.result == 'success' || needs.on-host-test.result == 'success') }}
+      - name: Fail
+        if: needs.test-results.result != 'skipped' && needs.test-results.result != 'success'
```

Comment:
Is there a reason why you have this check on this step level vs on the job level?  We end up with this no-op job running all the time (see attached image).  
<img width="379" alt="Screenshot 2025-02-04 at 2 04 25 PM" src="https://github.com/user-attachments/assets/391238eb-bcbc-4299-8861-f467eec1d897" />

Reply:
Fixed